### PR TITLE
feat: Bring in gostream to the monorepo

### PR DESCRIPTION
**Pull request by MarcoPolo**
#2438 Builds on gostream. I think it would be good to bring it in to the monorepo.

### CODE REVIEW

**Review by marten-seemann — CHANGES_REQUESTED** [review]
This doesn't preserve Git history. Do you want me to use my repo consolidation script to do that?

File: p2p/gostream/conn.go
Comment (marten-seemann) [inline p2p/gostream/conn.go:1]: Should this go into `p2p/net` instead?

Comment (MarcoPolo) [reply p2p/gostream/conn.go:1]: Done.

**Comment by MarcoPolo** [comment]
I think this is simple enough that there isn't a ton to gain from the perspective of keeping history to help debug things.

**Comment by marten-seemann** [comment]
> I think this is simple enough that there isn't a ton to gain from the perspective of keeping history to help debug things. But if you want to run your script, you are welcome to do so.

Probably not, I'd still prefer to preserve history. I don't see a good reason not to.

**Pull request closed without merging**